cmd/direnv/direnv.go

```diff
@@ -0,0 +1,196 @@
+package direnv
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"regexp"
+	"strings"
+
+	"github.com/buildsafedev/bsf/cmd/styles"
+	"github.com/spf13/cobra"
+)
+
+var Direnv = &cobra.Command{
+	Use:   "direnv",
+	Short: "direnv initializes the direnv environment for the project",
+	Run: func(cmd *cobra.Command, args []string) {
+		err := generateEnvrc()
+		if err != nil {
+			fmt.Println(styles.ErrorStyle.Render("error: ", err.Error()))
+			os.Exit(1)
+		}
+		err = fetchGitignore()
+		if err != nil {
+			fmt.Println(styles.ErrorStyle.Render("error: ", err.Error()))
+			os.Exit(1)
+		}
+
+		if envVar != "" {
+			err = setDIrenv(envVar)
+			if err != nil {
+				fmt.Println(styles.ErrorStyle.Render("error: ", err.Error()))
+				os.Exit(1)
+			}
+		}
+
+		cmD := exec.Command("direnv", "allow")
+		err = cmD.Run()
+		if err != nil {
+			os.Exit(1)
+		}
+	},
+}
+
+var (
+	envVar string
+)
+
+func init() {
+	Direnv.Flags().StringVarP(&envVar, "env", "e", "", "set environment variable [key=value]")
+}
+
+func generateEnvrc() error {
+
+	if _, err := os.Stat(".envrc"); err == nil {
+
+		read, err := os.ReadFile(".envrc")
+		if err != nil {
+			return err
+		}
+
+		if !strings.Contains(string(read), "use flake bsf/.") {
+
+			file, err := os.OpenFile(".envrc", os.O_APPEND|os.O_WRONLY, 0644)
+			if err != nil {
+				return err
+			}
+
+			_, err = file.WriteString("\nuse flake bsf/.")
```

Comment:
why are we using `"\nuse flake bsf/."` over here, whereas `"use flake bsf/."` below on line 86?